src/app/icd/IcdManagementServer.cpp

```diff
@@ -12,18 +12,19 @@ Status IcdManagementServer::RegisterClient(PersistentStorageDelegate & storage,
                                            uint64_t monitored_subject, chip::ByteSpan key,
                                            Optional<chip::ByteSpan> verification_key, bool is_admin)
 {
-    IcdMonitoringTable table(storage, fabric_index, GetClientsSupportedPerFabric());
+    IcdMonitoringTable table(storage, fabric_index, GetClientsSupportedPerFabric(), mSessionKeyStore);
 
     // Get current entry, if exists
-    IcdMonitoringEntry entry;
+    IcdMonitoringEntry entry(mSessionKeyStore);
     CHIP_ERROR err = table.Find(node_id, entry);
     if (CHIP_NO_ERROR == err)
     {
         // Existing entry: Validate Key if, and only if, the ISD has NOT administrator permissions
         if (!is_admin)
         {
             VerifyOrReturnError(verification_key.HasValue(), InteractionModel::Status::Failure);
-            VerifyOrReturnError(verification_key.Value().data_equal(entry.key), InteractionModel::Status::Failure);
+            // TODO Specs issue #7519
+            // VerifyOrReturnError(verification_key.Value().data_equal(entry.key()), InteractionModel::Status::Failure);
```

Comment:
This going to cause several tests to fail.

Reply:
Test are run with admin privilege.

Reply:
at least in CI. Will be fixed soon